### PR TITLE
Borrow&Request booklist and detailed views

### PR DESCRIPTION
**Pull request by IsaakC**
to reuse code I created a book_detail_common xml for the stuff all detailed views use

I changed the BookListAdapter to take an argument from the fragment to determine which onClick activity to do

### CODE REVIEW

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java [hunk 1]
```diff
@@ -68,21 +71,64 @@ protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position
         bookListViewHolder.author.setText(book.getAuthor());
         bookListViewHolder.isbn.setText(book.getIsbn());
 
-        book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+        switch(this.caller) {
+            case R.id.my_books:
+                book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+
+                holder.itemView.setOnClickListener(new View.OnClickListener()
```
Comment (huddy987) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java:78]: it seems backwards to be continually setting the onclick listener here by overriding onclick. at runtime we will be continually setting the onClick listener for no reason. I think this is done elsewhere in the project so we should talk about this during the sprint.

**Review by huddy987 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java [hunk 2]
```diff
@@ -68,21 +71,64 @@ protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position
...
+                book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+
+                holder.itemView.setOnClickListener(new View.OnClickListener()
+                {
+                    // Handles a click on an item in the recycler view
+                    @Override
+                    public void onClick(View v)
+                    {
+                        // Opens the book in detail view
+                        MyBooksDetailViewFragment bookDetailFragment = new MyBooksDetailViewFragment();
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java:85]: All of these functions are essentially the same. Is it possible to extract to a function and pass in a parameter of type Fragment?

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestDetailViewFragment.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import android.widget.TextView;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+
+/**
+ * Holds the view for seeing details on a book in the Requested tab
+ * The user will be able to interact with request options on the book
+ */
+//TODO: add status specific buttons, functions, and labels
+public class RequestDetailViewFragment extends Fragment
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestDetailViewFragment.java:23]: RequestDetailViewFragment and BorrowDetailViewFragment are very similar to each other and MyBooksDetailViewFragment. Is there any way to inherit from MyBooksDetailViewFragment or perhaps make a RequestsDetailViewFragment and inherit from that?

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java [hunk 4]
```diff
@@ -51,23 +66,86 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-
         return view;
     }
 
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState)
+    {
+        //TODO: actually query the books
+        FirebaseFirestore rootRef = FirebaseFirestore.getInstance();
+        Query query = rootRef.collection("Books").orderBy("title");
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:77]: Don't use hardcoded strings

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java [hunk 5]
```diff
@@ -51,23 +66,86 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        requestsRecyclerView = (RecyclerView) view.findViewById(R.id.requests_recycler_view);
+        requestsRecyclerView.setHasFixedSize(true);
+
+        requestsRecyclerLayoutManager = new LinearLayoutManager(this.getContext());
+        requestsRecyclerView.setLayoutManager(requestsRecyclerLayoutManager);
+
+        requestsRecyclerAdapter = new BookListAdapter(this.getContext(), options, R.id.requests);
+        requestsRecyclerView.setAdapter(requestsRecyclerAdapter);
+
+        Query query2 = rootRef.collection("Books").orderBy("title");
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:92]: Hardcoded strings again

**Review by KateVlaar — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java (hunk 5 above)
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:92]: In the commit I just pushed into master I added the collection to the strings.xml so you can use that

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java (hunk 1 above)
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java:78]: I take this comment back. Overriding onClick continually is necessary

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestDetailViewFragment.java (hunk 3 above)
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestDetailViewFragment.java:23]: This is a good idea and will make using the filter really easy in the other activities

**Pull request closed without merging**